tools/autograd/templates/ViewFuncs.h

```diff
@@ -0,0 +1,149 @@
+#pragma once
+
+// ${generated_comment}
+
+#include <torch/library.h>
+
+#include <c10/core/SymIntArrayRef.h>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Operators.h>
+#else
+$ops_headers
+#endif
+
+namespace torch { namespace autograd {
+
+// forward declaration so it can be a friend class of ViewFunc
+struct ViewFuncSavedStateGuard;
```

Comment:
nit: You may consider having non-generated code outside of the template file, so that when you make modifications to them it is not necessary to rebuild all of the generated code as well